### ISSUE TITLE
Unpack error

### ISSUE THREAD

**Issue by BARNZ** [hfig/MAPI#11]
Thanks for this library, crazy that there's not much stuff available for parsing .msg files.

I ran into a small problem today for certain emails. I'm getting the following php error:

**ErrorException**
Error: unpack(): Type V: not enough input, need 4, have 0
 **vendor\hfig\mapi\src\MAPI\Property\PropertyStore.php:140**

In the bit of code here - the second `unpack` call is failing because the `substr` function returns `false`
```php
if ($named) {
    $str_off = unpack('V', $rawProp)[1];
    $len     = unpack('V', substr($namesData, $str_off, 4))[1];
    $data = substr($namesData, $str_off + 4, $len);
    $prop = mb_convert_encoding($data, 'UTF-8', 'UTF-16LE');
}
```

I have dumped the values leading up to the error in this block:

![image](https://user-images.githubusercontent.com/2202345/87501981-8cfc9d80-c6b4-11ea-8369-3da3853d169e.png)

Which are as follows (respectively):

![image](https://user-images.githubusercontent.com/2202345/87502014-a867a880-c6b4-11ea-93a1-31258123cc1d.png)
![image](https://user-images.githubusercontent.com/2202345/87502030-b1587a00-c6b4-11ea-8dcc-363b05728114.png)
![image](https://user-images.githubusercontent.com/2202345/87502038-b4ec0100-c6b4-11ea-8696-36d1531bd4ea.png)

I don't really understand what's going on here, but if I follow your above code and `break` out of this loop when `substr ` returns `false` I can get around it:
![image](https://user-images.githubusercontent.com/2202345/87502340-773ba800-c6b5-11ea-950c-28a1ecb96bd8.png)

- Do you know what the problem could be here?
- Should I submit a PR to add this `false` check?

I could send you the problem email directly, I'm reluctant to post publicly on github.

**Comment by hfig** [comment]
I'm not very familiar with the file type, hence why the library is, by its own admission, a merely an enhanced port of a Ruby library. You can view the equivalent code it's ported from at https://github.com/aquasync/ruby-msg/blob/153b89c5a6c5d93c68d2434aed32725dc9ad7d3b/lib/mapi/msg.rb#L197  Specifically

```ruby
str_off = str.unpack('V').first
len = names_data[str_off, 4].unpack('V').first
Ole::Types::FROM_UTF16.iconv names_data[str_off + 4, len]
```

This segment of code is reading document property names. They're stored in one blob-like packed field, with a length for each property name and and then its data. Your MSG file apparently asks for a a property name stored over 2 billion bytes into the field. So something isn't right.

Rather than just check for a failure of ``substr()`` I think the best fix is to add a check for a bad ``$str_off`` directly. ie check if ``$str_off + 4 > strlen($namesData)`` and if so, skip further processing the property. Add a logger debug entry when you do this.

Submit a PR to that effect and I'll merge it.

**Comment by risototh** [comment]
I have this fix in the PR #22
